mautic/mautic
# [UX] Search again your last search

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

Dependency:
- #13689 

Users often need to repeat their last search but currently have to re-enter the same criteria manually, which is time-consuming and frustrating. 

This PR introduces a new button that appears after performing a search, enabling users to quickly redo their last search with a single click. This enhancement is tailored for each list, ensuring that the search context is preserved. By streamlining this process, we make the user experience more efficient and user-friendly, benefiting anyone who frequently uses the search function.

Features:
- It disappears if you didn't search anything yet
- It's individual per user
- It comes back even if you log out and log in again
- Don't use cookies nor session storage
- this uses 'localStorage' to store the search query, as alternative for cookies. That also means the query will disappear if you clear 'cookies and data' in the browser.

![image](https://github.com/mautic/mautic/assets/116097999/f965e608-9a39-4754-a086-35896207664f)







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Contacts 
3. Search anything and click enter
4. See the label
5. Go to Segments
6. See it's not there
7. Search another thing
8. See it appears
